File: frontend/lib/models/platform.dart
class Platform {
  final int id;
  final String name;
  final String displayName;
  final String baseUrl;
  final String logoUrl;

  Platform({
    required this.id,
    required this.name,
    required this.displayName,
    required this.baseUrl,
    this.logoUrl = '',
  });

  factory Platform.fromJson(Map<String, dynamic> json) {
    return Platform(
      id: json['id'],
      name: json['name'] ?? '',
      displayName: json['display_name'] ?? '',
      baseUrl: json['base_url'] ?? '',
      logoUrl: json['logo_url'] ?? '',
    );
  }
}

File: backend/app/schemas/platform.py
import datetime
from pydantic import BaseModel


class PlatformResponse(BaseModel):
    id: int
    name: str
    display_name: str
    base_url: str
    logo_url: str
    is_active: bool
    created_at: datetime.datetime

    model_config = {"from_attributes": True}


class PlatformDetail(PlatformResponse):
    login_config_json: str
    purchase_config_json: str
